crates/wysiwyg/src/tests/test_to_markdown.rs

```diff
@@ -203,6 +203,19 @@ fn list_ordered_and_unordered() {
     );
 }
 
+#[test]
+fn mention() {
+    assert_to_md_no_roundtrip(
+        r#"<a href="https://matrix.to/#/@test:example.org">test</a>"#,
+        r#"test"#,
```

Comment:
I think there will be some separate work to get the markdown output right for rooms (using the room ID like in your screenshot), related to https://github.com/matrix-org/matrix-rich-text-editor/pull/699#discussion_r1211824443.

For user mentions though, looking at the example in [the spec](https://spec.matrix.org/v1.6/client-server-api/#user-and-room-mentions), the markdown just contains the display name. So based on that I think we are ok here?

Reply:
I hadn't seen that example, I think you're right, we're ok here.

It's possible I'm confused as to the behaviour, after a previous discussion I thought room/user mentions behaved similarly but that may not be the case (although that seems odd to me)

Reply:
Yeah don't know where the discussions over display names went, but there is a strong argument that the raw text should be `@mxid`, just because with names changing, an application using the standard non formatted body will have a pretty hard time retrieving a user from a past display name, whereas mxid is way more consistent. (on the other hand if it uses the non formatted body, chances are it is just a basic client that doesn't transform the raw text further, but still 🤔 )

Reply:
I think you have a point @aringenbach. Having quickly looked at [MSC3952 - intentional mentions](https://github.com/matrix-org/matrix-spec-proposals/blob/main/proposals/3952-intentional-mentions.md) I think this may have been addressed there.

The current spec does seem to be explicit about it though:

> * When mentioning users, use the user’s potentially ambiguous display name for the anchor’s text. If the user does not have a display name, use the user’s ID.
> * When mentioning rooms, use the canonical alias [fallling back to alias/room ID] for the room. In all cases, use the alias/room ID being linked to as the anchor’s text.
>
> The text component of the anchor should be used in the event’s `body` where the mention would normally be represented, as shown in the example above.


Leaving us with the following

## User mention

```json
{
  "body": "Hello Alice!",
  "msgtype": "m.text",
  "format": "org.matrix.custom.html",
  "formatted_body": "Hello <a href='https://matrix.to/#/@alice:example.org'>Alice</a>!"
}
```

## Room mention

```json
{
  "body": "Hello #room:example.org!",
  "msgtype": "m.text",
  "format": "org.matrix.custom.html",
  "formatted_body": "Hello <a href='https://matrix.to/#/#room:example.org'>#room:example.org</a>!"
}
```